packages/browser-utils/src/encode-data.ts

```diff
@@ -0,0 +1,32 @@
+import { encode, decode } from 'js-base64';
+import { html, unsafeHtml, HtmlSafeString } from '@prairielearn/html';
+
+/**
+ * Use this function as an HTML component encode data that will be passed to the client.
+ *
+ * @param data The data to encode.
+ * @param elementId The element ID to use for the encoded data.
+ *
+ */
+export function EncodedData<T = unknown>(data: T, elementId: string): HtmlSafeString {
+  const encodedData = unsafeHtml(encode(JSON.stringify(data)));
+  return html`<script id="${elementId}" type="application/base64">
+    ${encodedData}
+  </script>`;
+}
+
+/**
+ * Decode data that was passed to the client from in HTML component using EncodeData().
+ *
+ * @param elementId The element ID that stores the encoded data, from from EncodedData().
+ * @returns The decoded data.
+ */
+export function decodeData<T = any>(elementId: string): T {
```

Comment:
Do we want to take this opportunity to add zod checking to the decode? I haven't yet encountered a bug due to mismatched types between the server and client but I think it would be pretty ugly if it happened.

Reply:
I'll pass on that for now, but it's a good future enhancement!